sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/AlterTableAddPartitionSuite.scala

```diff
@@ -44,4 +44,13 @@ class AlterTableAddPartitionSuite
       checkPartitions(t, Map("p1" -> ""))
     }
   }
+
+  test("SPARK-34143: add a partition to fully partitioned table") {
+    withNamespaceAndTable("ns", "tbl") { t =>
+      sql(s"CREATE TABLE $t (p0 INT, p1 STRING) $defaultUsing PARTITIONED BY (p0, p1)")
```

Comment:
is it allowed in v1 tables?

Reply:
+1 for @cloud-fan 's question. Yes, this is not allowed in DSv1.
```
spark-sql> create table t(a STRING, b STRING) using orc partitioned by (a,b);
Error in query: Cannot use all columns for partition columns
```

Reply:
I think, v2 impl should decide allowed it or not.

Reply:
sounds reasonable in general, as we tend to move limitations to the catalog implementation. @rdblue @stczwd what do you think?

Reply:
Currently, we do not disallow it so I'm fine to fix the testing v2 catalog implementation. But we should figure out if we should allow fully partitioned table or not later.

Reply:
I think it's weird to support a table that doesn't have any data column. But looks like we're currently exposing it via the interface and letting users to decide, right? I am okay with this change - it's a test-only for the current status in the end.

Reply:
> Currently, we do not disallow it so I'm fine to fix the testing v2 catalog implementation. But we should figure out if we should allow fully partitioned table or not later.

I can not find any user cases with full partitioned tables, maybe we should disallow it?

Reply:
I could imagine the case when fully partitioned table could speed up table creation. Let's image we have heavy data partitioned by some meta-data columns like date, id and etc. We need a table to analyse only meta-data, and don't want to spend time on parsing/schema inferring/reading data columns. So, we could quickly create a table fully partitioned only by meta-data columns, and don't waste time on data columns.

Reply:
Since disallowing it would be a breaking change and we should discuss it separately. I'm merging this PR first to fix the testing implementation.